ext/dom/document.c

```diff
@@ -1999,18 +2002,21 @@ static void _dom_document_schema_validate(INTERNAL_FUNCTION_PARAMETERS, int type
 	}
 #endif
 
-	PHP_LIBXML_SANITIZE_GLOBALS(validate);
-	xmlSchemaSetValidOptions(vptr, valid_opts);
-	xmlSchemaSetValidErrors(vptr, php_libxml_error_handler, php_libxml_error_handler, vptr);
-	is_valid = xmlSchemaValidateDoc(vptr, docp);
-	xmlSchemaFree(sptr);
-	xmlSchemaFreeValidCtxt(vptr);
-	PHP_LIBXML_RESTORE_GLOBALS(validate);
+	if (vptr) {
+		PHP_LIBXML_SANITIZE_GLOBALS_DECL(validate);
+		PHP_LIBXML_SANITIZE_GLOBALS(validate);
```

Comment:
VC11 even stumbled over 2 `PHP_LIBXML_SANITIZE_GLOBALS_DECL` after another, so I had to create a new block for `PHP_LIBXML_SANITIZE_GLOBALS_DECL(validate)`

`if (vptr)` always evaluates to true here, because the is a check om !vptr with a return a few lines back.